test/unit/services/awsAdapter_spec.js

```diff
@@ -148,4 +150,88 @@ describe("The AwsAdapter class ", function () {
 			});
 		});
 	});
-});
+
+	describe("gets properties of given instance", function () {
+		var describeInstancesStub;
+		var result;
+		before(function () {
+			var awsAdapter = new AwsAdapter(ec2);
+			describeInstancesStub = Sinon.stub(ec2, "describeInstancesAsync", function () {
+				var data = { Reservations : [ { Instances : [ { PublicIpAddress : "127.0.0.1" } ] } ] };
+				return Bluebird.resolve(data);
+			});
+			awsAdapter.describeInstance(VALID_EC2_INSTANCE)
```

Comment:
I think this before block is prone to timing errors since the promise chain is never returned.